comptests/tests/main_test.go

```diff
@@ -0,0 +1,113 @@
+// +build comptest
+
+package main
+
+import (
+	"context"
+	"log"
+	"testing"
+
+	"github.com/ONSdigital/dp-find-insights-poc-api/comptests"
+	"github.com/ONSdigital/dp-find-insights-poc-api/model"
+	"github.com/jackc/pgx/v4"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+/*
+example code for comptest with INSERT and ROLLBACK
+
+Also see pkg/geodata/rowquery_test.go which uses
+
+		db.DB().Exec("BEGIN")
+		defer db.DB().Exec("ROLLBACK")
+
+*/
+
+const dsn = "postgres://insights:insights@localhost:54322/censustest"
+
+var db *gorm.DB
+
+func init() {
+	comptests.SetupDockerDB(dsn)
+	model.SetupDB(dsn)
+	var err error
+	db, err = gorm.Open(postgres.Open(dsn), &gorm.Config{})
+	if err != nil {
+		log.Print(err)
+	}
+
+}
+
+func TestGorm(t *testing.T) {
```

Comment:
is this a test that gorm respects the inner transaction rollback stuff we're using to insulate tests?

Reply:
It's an example of how to use gorm transactions.

Reply:
ahh gotcha - maybe TestGormExample or something would be clearer? Also, what does it mean if either of these tests fail?

Reply:
see comments below